jina/drivers/index.py

```diff
@@ -27,14 +27,14 @@ def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
             self.pea.logger.warning(f'these bad docs can not be added: {bad_docs}')
 
         if docs_pts:
-            self.exec_fn(np.array([hash(doc.id) for doc in docs_pts]), np.stack(embed_vecs))
+            self.exec_fn(np.array([int(doc.id) for doc in docs_pts]), np.stack(embed_vecs))
 
 
 class KVIndexDriver(BaseIndexDriver):
     """Serialize the documents/chunks in the request to key-value JSON pairs and write it using the executor
     """
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        keys = [hash(doc.id) for doc in docs]
+        keys = [int(doc.id) for doc in docs]
```

Comment:
does this have to be done in the VectorIndexDriver as well?

Reply:
It is. See line 30.

Reply:
ah okay, I searched for VectorIndexDriver and did not find a hit. 😳